api/src/Entity/Activity.php

```diff
@@ -3,48 +3,84 @@
 namespace App\Entity;
 
 use ApiPlatform\Core\Annotation\ApiResource;
-use App\Repository\ActivityRepository;
+use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\ORM\Mapping as ORM;
 
 /**
- * @ORM\Entity(repositoryClass=ActivityRepository::class)
+ * @ORM\Entity()
  */
 #[ApiResource]
-class Activity extends BaseEntity {
+class Activity extends AbstractContentNodeOwner implements BelongsToCampInterface {
     /**
-     * @ORM\Column(type="text")
+     * @ORM\OneToMany(targetEntity="ScheduleEntry", mappedBy="activity", orphanRemoval=true)
+     *
+     * @var ScheduleEntry[]
      */
-    public ?string $title = null;
+    public $scheduleEntries;
 
     /**
-     * @ORM\Column(type="text")
+     * @ORM\OneToMany(targetEntity="ActivityResponsible", mappedBy="activity", orphanRemoval=true)
+     *
+     * @var ActivityResponsible[]
      */
-    public string $location = '';
+    public $activityResponsibles;
 
     /**
-     * @ORM\ManyToOne(targetEntity=Camp::class, inversedBy="activities")
-     * @ORM\JoinColumn(nullable=false)
+     * @ORM\ManyToOne(targetEntity="Camp", inversedBy="activities")
+     * @ORM\JoinColumn(nullable=false, onDelete="cascade")
      */
     public ?Camp $camp = null;
 
     /**
-     * @ORM\OneToOne(targetEntity=ContentNode::class, mappedBy="owner", cascade={"persist", "remove"})
+     * @ORM\ManyToOne(targetEntity="Category")
+     * @ORM\JoinColumn(nullable=false)
      */
-    public ?ContentNode $rootContentNode = null;
+    public ?Category $category = null;
 
-    public function setRootContentNode(?ContentNode $rootContentNode): self {
-        // unset the owning side of the relation if necessary
-        if (null === $rootContentNode && null !== $this->rootContentNode) {
-            $this->rootContentNode->owner = null;
-        }
+    /**
+     * @ORM\Column(type="text")
+     */
+    public ?string $title = null;
 
-        // set the owning side of the relation if necessary
-        if (null !== $rootContentNode && $rootContentNode->owner !== $this) {
-            $rootContentNode->owner = $this;
-        }
+    /**
+     * @ORM\Column(type="text")
+     */
+    public string $location = '';
 
-        $this->rootContentNode = $rootContentNode;
+    public function __construct() {
+        $this->scheduleEntries = new ArrayCollection();
+        $this->activityResponsibles = new ArrayCollection();
+    }
+
+    public function getCamp(): ?Camp {
+        return $this->camp;
+    }
+
+    public function getScheduleEntries(): array {
+        return $this->scheduleEntries->getValues();
+    }
+
+    public function addScheduleEntry(ScheduleEntry $scheduleEntry): void {
+        $scheduleEntry->activity = $this;
+        $this->scheduleEntries->add($scheduleEntry);
```

Comment:
Die von Symfony generierten Adders sehen etwa so aus:
```suggestion
        if (!$this->scheduleEntries->contains($scheduleEntry) {
            $this->scheduleEntries[] = $scheduleEntries;
            $scheduleEntry->activity = $this;
        }

        return $this;
```
Wollen wir diesen Stil adoptieren? Gibt es Nachteile wenn der "parent" erst nach dem adden zur Collection gesetzt wird? Macht es einen Unterschied ob wir `->add` oder `[] =` verwenden?